pkg/internal/testhelpers/containers.go

```diff
@@ -26,24 +31,28 @@ var (
 	pgPort nat.Port = "5432/tcp"
 )
 
-const defaultDB = "postgres"
-
-func pgConnectURL(t *testing.T, dbName string) string {
-	template := "postgres://postgres:password@%s:%d/%s"
-	return fmt.Sprintf(template, pgHost, pgPort.Int(), dbName)
+func pgConnectURL(dbName string) string {
+	return fmt.Sprintf(connectTemplate, pgHost, pgPort.Int(), dbName)
 }
 
-// WithDB establishes a database for testing and calls the callback
+// WithDB establishes a database for testing and calls the callback.
 func WithDB(t *testing.T, DBName string, f func(db *pgxpool.Pool, t *testing.T, connectString string)) {
 	db := dbSetup(t, DBName)
 	defer func() {
 		db.Close()
 	}()
-	f(db, t, pgConnectURL(t, DBName))
+	f(db, t, pgConnectURL(DBName))
+}
+
+func SetDatabaseToReadOnly(t testing.TB, db *pgxpool.Pool, DBName string) {
+	_, err := db.Exec(context.Background(), fmt.Sprintf("ALTER DATABASE %s SET default_transaction_read_only = true", DBName))
```

Comment:
I think you want to close all the pool connections after this to make sure new connections are started with this mode?

Reply:
Updated to use a read-only session for query integration tests.

Could you take another look?